components/automate-ui/src/app/modules/policy/add-members/policy-add-members.component.ts

```diff
@@ -376,16 +375,27 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.expressionForm.addControl('identityProvider', new FormControl('', Validators.required));
   }
 
-  private addNameControl(): void {
-    this.expressionForm.addControl('name', new FormControl('',
-      [
-        Validators.required,
-        Validators.pattern(Regex.patterns.NO_MIXED_WILDCARD)
-      ]
-    )
-    );
+  private addNameControl(identityProvider): void {
+    if (identityProvider === 'ldap' || identityProvider === 'saml' ) {
+        this.expressionForm.addControl('name', new FormControl('',
+          [
+            Validators.required,
+            Validators.pattern(Regex.patterns.NO_MIXED_WILDCARD_ALLOW_SPECIAL)
+          ]
+        )
+      );
+    } else {
+        this.expressionForm.addControl('name', new FormControl('',
+          [
+            Validators.required,
+            Validators.pattern(Regex.patterns.NO_MIXED_WILDCARD_ALLOW_HYPHEN)
```

Comment:
I'm not sure this regex is getting enforced. I can create local team, user, and token member expressions with special characters (like: `,BDFda&&&!!!___`)

Reply:
actually I just realized we also don't have the right validation on our membership API 😬

Reply:
thanks for looking!

Reply:
@bcmdarroch - I wonder if maybe I had a messed up version pushed up?  Will you give this a shot again.  All patterns are working for me as expected.

<img width="620" alt="Screen Shot 2019-11-26 at 10 13 20 AM" src="https://user-images.githubusercontent.com/16737484/69660715-bbe6d600-1035-11ea-8145-6b71d6b79acd.png">
<img width="617" alt="Screen Shot 2019-11-26 at 10 13 57 AM" src="https://user-images.githubusercontent.com/16737484/69660716-bc7f6c80-1035-11ea-8f06-00edc4117197.png">
<img width="621" alt="Screen Shot 2019-11-26 at 10 11 08 AM" src="https://user-images.githubusercontent.com/16737484/69660750-d1f49680-1035-11ea-922e-0d239e63b95a.png">
<img width="618" alt="Screen Shot 2019-11-26 at 10 10 40 AM" src="https://user-images.githubusercontent.com/16737484/69660763-d9b43b00-1035-11ea-9d4e-8fe5ee06210d.png">

Reply:
Just passing by and realized this is a very odd error message given the input (which is text, but does not contain a wildcard).

Reply:
@susanev - do you have an error message wording request or would you like me to use whatever we can come up with?

@msorens suggested the following above:
>"Use either a single wildcard or a name without colons or wildcards."
>That is very precise now, but probably not the greatest wording.

Reply:
let me check with docs folks next week, they have better skills with words than i do.